javacord-core/src/main/java/org/javacord/core/entity/AttachmentImpl.java

```diff
@@ -155,6 +168,31 @@ public Optional<Boolean> isEphemeral() {
         return Optional.ofNullable(ephemeral);
     }
 
+    @Override
+    public Optional<Float> getDurationSeconds() {
+        return Optional.ofNullable(durationSeconds);
+    }
+
+    @Override
+    public Optional<Byte[]> getWaveForm() {
+        if (waveform == null) {
+            return Optional.empty();
+        }
+
+        byte[] decoded = Base64.getDecoder().decode(waveform);
+        Byte[] bytes = new Byte[decoded.length];
+        for (int i = 0; i < decoded.length; i++) {
+            bytes[i] = decoded[i];
+        }
+        return Optional.of(bytes);
+    
```

Comment:
A byte array `byte[]` is not a primitive datatype and therefore can be used in generics, no need to use the wrapper class
```suggestion
    public Optional<byte[]> getWaveForm() {
        return Optional.ofNullable(waveform).map(wave -> Base64.getDecoder().decode(waveform));
```